packages/medusa/src/services/order-edit.ts

```diff
@@ -0,0 +1,51 @@
+import { EntityManager } from "typeorm"
+import { MedusaError } from "medusa-core-utils"
+import { OrderEditRepository } from "../repositories/order-edit"
+import { OrderEditStatus } from "../models/order-edit"
+import { TransactionBaseService } from "../interfaces"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  orderEditRepository: typeof OrderEditRepository
+}
+
+export class OrderEditService extends TransactionBaseService {
+  protected transactionManager_: EntityManager | undefined
+  protected readonly manager_: EntityManager
+
+  protected readonly orderEditRepository_: typeof OrderEditRepository
+
+  constructor({ manager, orderEditRepository }: InjectedDependencies) {
+    super({ manager, orderEditRepository })
```

Comment:
todo: `arguments[0]` instead to allow this class to be extender with other dependencies. otherwise, a use that would extend this class and if it is called with `withTransaction`, the returned instance will not keep the custom deps. part of the following issue https://linear.app/medusajs/issue/CORE-575/extending-class-with-custom-deps-taken-into-account